internal/ethapi/api.go

```diff
@@ -709,13 +709,17 @@ func (s *PublicBlockChainAPI) GetBlockByNumber(ctx context.Context, number rpc.B
 	if block != nil && err == nil {
 		response, err := s.rpcMarshalBlock(ctx, block, true, fullTx)
 
-		if err == nil && s.b.RPCEthCompatibility() {
-			addEthCompatibilityFields(ctx, response, s.b, block.Header())
-			if number == rpc.PendingBlockNumber {
-				// Pending blocks need to nil out a few fields
-				for _, field := range []string{"hash", "nonce", "miner"} {
-					response[field] = nil
+		if err == nil {
+			if s.b.RPCEthCompatibility() {
+				addEthCompatibilityFields(ctx, response, s.b, block.Header())
+				if number == rpc.PendingBlockNumber {
+					// Pending blocks need to nil out a few fields
+					for _, field := range []string{"hash", "nonce", "miner"} {
+						response[field] = nil
+					}
 				}
+			} else {
+				delete(response, "gasLimit")
```

Comment:
Wouldn't we want to keep the `gasLimit` field in the response irrespective of the `RPCEthCompatibility` flag?

Reply:
Yes, you are right. I thought we should not add it to keep the same result layout for users expecting no RPCEthCompatibility, but actually the point is to return exactly values which are included in the block hash. I will also have to check that the order is the same, otherwise the RPCEthCompatibility=off feature won't make any sense.

In general, I would be happy to remove the feature unless there is some known usage.